Document contrib and semantic

olafurpg opened the pull request.

olafurpg committed:
Document contrib and semantic

DavidDudson commented on readme/Contrib.scalatex:
```diff
@@ -0,0 +1,69 @@
+      trait B
+      object C
+      object D
+      """.find(_.is[Defn.Object])
```
is `is` and actual method? or is it just short for `isInstanceOf`

olafurpg replied:
It's an actual method, you can create custom "classifiers" like [here](https://github.com/scalameta/scalameta/blob/254ec8f81434214ed437f52213f852cd0a5069f1/scalameta/contrib/src/main/scala/scala/meta/contrib/TokenClasses.scala#L10-L16) and use the same syntax `.is[Trivia]`

DavidDudson commented on readme/Contrib.scalatex:
```diff
@@ -0,0 +1,69 @@
+
+  @p
+    Contrib has a @code{Equal} typeclass for comparing trees by structural or
+    syntactic equality.
```
This will all need to be changed very soon. I suggest waiting for the equality merge and using the updated api.

Something like you can compare trees!

`q"val x = 2".isEqual(q"def x = 2")`

If you really require it. you case also compare syntax

... Some syntax test

DavidDudson commented on readme/Contrib.scalatex:
```diff
@@ -0,0 +1,69 @@
+@import Main._
+@import scalaworld.Readme._
+
+@sect{Part 6 - Contrib}
```
Missing:

asTerm/asType

DavidDudson reviewed (changes requested):
I'm not happy merging with the old equality api. otherwise it is fine.

olafurpg committed:
Clean up repo a bit.

cc/ https://github.com/scalameta/scalameta/issues/787

- Remove a few unused scripts
- Setup CI with drone
- Setup publish on merge to master

olafurpg committed:
Use latest contrib api

olafurpg committed:
Try to fix Ci failure

olafurpg commented:
I setup CI, publish on merge, updated the scala.meta version in the ammonite import example and updated the equality examples to the latest contrib api.

There are still quite a few undocumented parts (Mirror, contrib, scalahost) and the repo is still has way too many leftovers from the old scalafix repo. However, I'll leave those for another PR. I hope it will be more rewarding to contribute docs here esp. with publish on merge setup.

xeno-by commented on build.sbt:
```diff
@@ -4,6 +4,10 @@ import sbt.ScriptedPlugin
 import sbt.ScriptedPlugin._
 import scoverage.ScoverageSbtPlugin.ScoverageKeys._
 
+lazy val MetaVersion     = "1.6.0"
```
How about 1.7.0?

xeno-by commented on build.sbt:
```diff
@@ -4,6 +4,10 @@ import sbt.ScriptedPlugin
 import scoverage.ScoverageSbtPlugin.ScoverageKeys._
 
+lazy val MetaVersion     = "1.6.0"
+lazy val ParadiseVersion = "3.0.0-M7"
```
And M8?

olafurpg commented on core/src/main/scala/scalaworld/Versions.scala:
```diff
@@ -2,6 +2,8 @@ package scalaworld
-  val scala   = "2.11.8"
+  val stable = nightly
+  val scala = "2.11.10"
+  val scalameta = "1.7.0"
```
See here @xeno-by

xeno-by replied:
Got it, thanks!

DavidDudson commented:
@olafurpg Can you push this to GH-pages, the site is obviously pointing to the old version

olafurpg commented:
I was working on the auto publish docs from ci on merge yesterday. Will
hopefully get it working today, I'm too stubborn to publish the docs
manually before I get it working :p
On Sun, 16 Apr 2017 at 08:25, David Dudson <notifications@github.com> wrote:

> @olafurpg <https://github.com/olafurpg> Can you push this to GH-pages,
> the site is obviously pointing to the old version
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/scalameta/tutorial/pull/9#issuecomment-294336238>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABV8XRn0IdZWBc1WF0sHCexlwq3yyqZqks5rwbRCgaJpZM4MOnG_>
> .
>

olafurpg commented:
The page has now been updated. I've never been happy with the `update-gh-pages` bash script, now I finally found a cleaner way to do it with sbt-site and sbt-ghpages :)

olafurpg commented:
Awww, jiss. Auto-publish docs on merge is finally working using a deploy key https://platform-ci.scala-lang.org/scalameta/tutorial/28 Deploy keys are nice since they only have access to a single repo and can easily be revoked if exposed
